### PR TITLE
BeforeSourcePath should point to base commit

### PR DESCRIPTION
**Pull request by jpraet**
Clicking "View File" on a deleted file gives a 404 because the link points to head commit instead of base commit.

### CODE REVIEW

File: routers/web/repo/compare.go [hunk 1]
```diff
@@ -78,8 +78,8 @@ func setPathsCompareContext(ctx *context.Context, base, head *git.Commit, headOw
 	ctx.Data["SourcePath"] = SourceCommitURL(headOwner, headName, head)
 	ctx.Data["RawPath"] = RawCommitURL(headOwner, headName, head)
 	if base != nil {
-		ctx.Data["BeforeSourcePath"] = SourceCommitURL(headOwner, headName, head)
-		ctx.Data["BeforeRawPath"] = RawCommitURL(headOwner, headName, head)
+		ctx.Data["BeforeSourcePath"] = SourceCommitURL(headOwner, headName, base)
```
Comment (jpraet) [inline routers/web/repo/compare.go:81]: Note sure if headOwner and headName are correct here either?

Comment (lunny) [reply routers/web/repo/compare.go:81]: `base` is the pull request start point commit. It should be in base repository and head repository. `head` is the last commit of this pull request. So I think your change should be right but I haven't test it.

**Review by KN4CK3R — APPROVED** [review]

**Comment by KN4CK3R** [comment]
That's a regression of #17551 and the cause of #18861.

**Review by silverwind — APPROVED** [review]
Can confirm it fixes https://github.com/go-gitea/gitea/issues/18861.

**Comment by lunny** [comment]
Please send backport

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**